rinja_derive/src/tests.rs

```diff
@@ -569,7 +569,10 @@ fn check_bool_conditions() {
     );
     compare(
         "{% if y == 3 || (x == 12 || true) %}{{x}}{% endif %}",
-        r"if *(&(self.y == 3 || (self.x == 12 || true)) as &rinja::helpers::core::primitive::bool) {
+        r"
+if rinja::helpers::as_bool(&(self.y == 3))
```

Comment:
Same in this case, it's a comparison so we already know for sure it's a boolean.